pkg/agent/controller/networkpolicy/networkpolicy_controller_test.go

```diff
@@ -355,40 +355,42 @@ func TestAddNetworkPolicyWithMultipleRules(t *testing.T) {
-			}
-			if !actualRule.FromAddresses.Equal(desiredRule1.FromAddresses) {
-				t.Errorf("Expected FromAddresses %v, got %v", actualRule.FromAddresses, desiredRule1.FromAddresses)
-			}
-			if !actualRule.ToAddresses.Equal(desiredRule1.ToAddresses) {
-				t.Errorf("Expected ToAddresses %v, got %v", actualRule.ToAddresses, desiredRule1.ToAddresses)
-			}
-			if !actualRule.Pods.Equal(desiredRule1.Pods) {
-				t.Errorf("Expected Pods %v, got %v", actualRule.Pods, desiredRule1.Pods)
+	for i := 0; i < 2; i++ {
+		select {
+		case ruleID := <-reconciler.updated:
+			actualRule, _ := reconciler.getLastRealized(ruleID)
+			if actualRule.Direction == v1beta1.DirectionIn {
+				if !assert.ElementsMatch(t, actualRule.Services, desiredRule1.Services) {
+					t.Errorf("Expected Services %v, got %v", actualRule.Services, desiredRule1.Services)
+				}
+				if !actualRule.FromAddresses.Equal(desiredRule1.FromAddresses) {
+					t.Errorf("Expected FromAddresses %v, got %v", actualRule.FromAddresses, desiredRule1.FromAddresses)
+				}
+				if !actualRule.ToAddresses.Equal(desiredRule1.ToAddresses) {
+					t.Errorf("Expected ToAddresses %v, got %v", actualRule.ToAddresses, desiredRule1.ToAddresses)
+				}
+				if !actualRule.Pods.Equal(desiredRule1.Pods) {
+					t.Errorf("Expected Pods %v, got %v", actualRule.Pods, desiredRule1.Pods)
+				}
 			}
-		}
-		if actualRule.Direction == v1beta1.DirectionOut {
-			if !assert.ElementsMatch(t, actualRule.Services, desiredRule2.Services) {
-				t.Errorf("Expected Services %v, got %v", actualRule.Services, desiredRule2.Services)
-			}
-			if !actualRule.FromAddresses.Equal(desiredRule2.FromAddresses) {
-				t.Errorf("Expected FromAddresses %v, got %v", actualRule.FromAddresses, desiredRule2.FromAddresses)
-			}
-			if !actualRule.ToAddresses.Equal(desiredRule2.ToAddresses) {
-				t.Errorf("Expected ToAddresses %v, got %v", actualRule.ToAddresses, desiredRule2.ToAddresses)
-			}
-			if !actualRule.Pods.Equal(desiredRule2.Pods) {
-				t.Errorf("Expected Pods %v, got %v", actualRule.Pods, desiredRule2.Pods)
+			if actualRule.Direction == v1beta1.DirectionOut {
+				if !assert.ElementsMatch(t, actualRule.Services, desiredRule2.Services) {
+					t.Errorf("Expected Services %v, got %v", actualRule.Services, desiredRule2.Services)
+				}
+				if !actualRule.FromAddresses.Equal(desiredRule2.FromAddresses) {
+					t.Errorf("Expected FromAddresses %v, got %v", actualRule.FromAddresses, desiredRule2.FromAddresses)
+				}
+				if !actualRule.ToAddresses.Equal(desiredRule2.ToAddresses) {
+					t.Errorf("Expected ToAddresses %v, got %v", actualRule.ToAddresses, desiredRule2.ToAddresses)
+				}
+				if !actualRule.Pods.Equal(desiredRule2.Pods) {
+					t.Errorf("Expected Pods %v, got %v", actualRule.Pods, desiredRule2.Pods)
+				}
 			}
-		}
 
-	case <-time.After(time.Millisecond * 100):
-		t.Fatal("Expected two updates, got timeout")
+		case <-time.After(time.Millisecond * 100):
+			t.Fatal("Expected first rule update, got timeout")
```

Comment:
No `first` IMO.

Reply:
Nice catch, thanks. Will update in the next commit.